specs/phase1/validator.md

```diff
@@ -0,0 +1,520 @@
+
+Then:
+* Set `light_client_bits = best_aggregate.aggregation_bits`
+* Set `light_client_signature = best_aggregate.signature`
+
+```python
+def get_best_light_client_aggregate(block: BeaconBlock,
+                                    aggregates: Sequence[LightClientVote]) -> LightClientVote:
+    viable_aggregates = [
+        aggregate for aggregate in aggregates
+        if aggregate.slot == compute_previous_slot(block.slot) and aggregate.beacon_block_root == block.parent_root
+    ]
+
+    return max(
+        viable_aggregates,
+        key=lambda a: len([i for i in a.aggregation_bits if i == 1]),
+        default=LightClientVote(),
+    )
+```
+
+#### Packaging into a `SignedBeaconBlock`
+
+Packaging into a `SignedBeaconBlock` is unchanged from Phase 0.
+
+### Attesting
+
+A validator is expected to create, sign, and broadcast an attestation during each epoch.
+
+Assignments and the core of this duty are unchanged from Phase 0. There are a few additional fields related to the assigned shard chain and custody bit.
+
+The `Attestation` and `AttestationData` defined in the [Phase 1 Beacon Chain spec]() utilizes `shard_transition_root: Root` rather than a full `ShardTransition`. For the purposes of the validator and p2p layer, a modified `FullAttestationData` and containing `FullAttestation` are used to send the accompanying `ShardTransition` in its entirety. Note that due to the properties of SSZ `hash_tree_root`, the root and signatures of `AttestationData` and `FullAttestationData` are equivalent.
+
+#### `FullAttestationData`
+
+```python
+class FullAttestationData(Container):
+    slot: Slot
+    index: CommitteeIndex
+    # LMD GHOST vote
+    beacon_block_root: Root
+    # FFG vote
+    source: Checkpoint
+    target: Checkpoint
+    # Current-slot shard block root
+    head_shard_root: Root
+    # Full shard transition
+    shard_transition: ShardTransition
+```
+
+#### `FullAttestation`
+
+```python
+class FullAttestation(Container):
+    aggregation_bits: Bitlist[MAX_VALIDATORS_PER_COMMITTEE]
+    data: FullAttestationData
+    custody_bits_blocks: List[Bitlist[MAX_VALIDATORS_PER_COMMITTEE], MAX_SHARD_BLOCKS_PER_ATTESTATION]
+    signature: BLSSignature
+```
+
+#### Timing
+
+Note the timing of when to create/broadcast is altered from Phase 1.
+
+A validator should create and broadcast the `attestation` to the associated attestation subnet when either (a) the validator has received a valid `BeaconBlock` from the expected beacon block proposer and a valid `ShardBlock` for the expected shard block proposer for the assigned `slot` or (b) one-half of the `slot` has transpired (`SECONDS_PER_SLOT / 2` seconds after the start of `slot`) -- whichever comes _first_.
+
+#### Attestation data
+
+`attestation_data` is constructed in the same manner as Phase 0 but uses `FullAttestationData` with the addition of two fields -- `head_shard_root` and `shard_transition`.
```

Comment:
Validators actual contrsuct a `FullAttestationData` that has a the full `shard_transition` rather than the `root`. This is sent on the wire (will note in validator and p2p spec). The block producers compress the `shard_transition`s into `root`s and only including winning `ShardTransition` objects separately

Reply:
Oops, I missread the sentence and thought it was referring to `attestation_data`. My b!